### ISSUE TITLE
Add new datasets

### ISSUE THREAD

**Issue by kasnerz** [kasnerz/tabgenie#10]
If there is any dataset which is not added yet and has suitable format, add it.

Potential candidates (not checked yet):

- [ ] [Turning Tables](https://github.com/oriyor/turning_tables) -- QA-focused
- [x] [WikiSQL](https://github.com/salesforce/WikiSQL)
- [ ] [WikiTableT](https://github.com/mingdachen/WikiTableT) -- too big
- [x] [WikiTableText](https://www.tensorflow.org/datasets/catalog/wiki_table_text)
- [x] [CACAPO](https://github.com/TallChris91/CACAPO-Dataset)
- [x] [EventNarrative](https://www.kaggle.com/datasets/acolas1/eventnarration)
- [ ] [AGENDA](https://github.com/rikdz/GraphWriter) -- noisy

Always check the licenses!

If you want to add a dataset to the Hugginface datasets, see the comment below: https://github.com/kasnerz/tabgenie/issues/10#issuecomment-1403417267

**Comment by oplatek** [comment]
@kasnerz do you understand how CACAPO created the alignments and if we can create explicit alignments in the "webnlg+" XML format that CACAPO uses?

The licensing is unfortunate (but CC0 for CACAPO "thin dataset" is nice).

**Comment by kasnerz** [comment]
@oplatek From the paper:
> Sentences were manually aligned with data using Prodigy9, a data annotation tool (Montani and Honnibal, 2018), in a
attribute-value pair format, done by two expert annotators. The annotators annotated a part of the dataset jointly (1,755 sentences), resulting in Cohen’s κ = 0.67 (substantial agreement; Landis and Koch, 1977) and a 70.92% agreement. This agreement was deemed high enough for a single coder per item approach for the rest of the dataset.

When the dataset came out, I asked the authors for the full version and they sent it to me. 

But now I see that it's actually publicly available under a **CC-BY-4.0 license**, which is definitely less restrictive than stated in the paper:
https://dataverse.nl/dataset.xhtml?persistentId=doi:10.34894/LIBYHP

![screen-2023-01-12-15-15-40](https://user-images.githubusercontent.com/8361334/212090222-768b8d55-cae6-4fbc-8550-5fd3b70d4092.png)

**Comment by kasnerz** [comment]
Here is a sample code which enables parsing an external dataset (LogicNLG in this case) and uploading it to Huggingface dataset:

**Commands:**
```bash
pip install huggingface_hub
huggingface-cli login
./logicnlg.py
```

**logicnlg.py** (created in the root directory of https://github.com/wenhuchen/LogicNLG )


```python
#!/usr/bin/env python3
"""LogicNLG: natural language generation with logical inference in the intermediate steps"""

import json
import datasets
import glob
import os

_CITATION = """\
@inproceedings{chen2020logical,
  title={Logical Natural Language Generation from Open-Domain Tables},
  author={Chen, Wenhu and Chen, Jianshu and Su, Yu and Chen, Zhiyu and Wang, William Yang},
  booktitle={Proceedings of the 58th Annual Meeting of the Association for Computational Linguistics},
  pages={7929--7942},
  year={2020}
}
"""
_DESCRIPTION = """\
LogicNLG: natural language generation with logical inference in the intermediate steps
"""

_URL = "https://github.com/wenhuchen/LogicNLG"

class LogicNLG(datasets.GeneratorBasedBuilder):
    def _info(self):
        return datasets.DatasetInfo(
            description=_DESCRIPTION,
            features=datasets.Features({
                'table': datasets.Value(dtype='large_string'),
                "ref": datasets.Value(dtype='string'),
                "linked_columns": datasets.Value(dtype='string'),
                "title": datasets.Value(dtype='string'),
                "template": datasets.Value(dtype='string'),
                "table_id": datasets.Value(dtype='string'),
            }),
            supervised_keys=None,
            homepage="https://wenhuchen.github.io/logicnlg.github.io/",
            citation=_CITATION,
        )

    def _split_generators(self, dl_manager):
        """Returns SplitGenerators."""
        return [
            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": "./data", "split" : "train"}),
            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": "./data", "split" : "dev"}),
            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": "./data", "split" : "test"}),
        ]

    def _generate_examples(self, filepath, split):
        filename = split if split != "dev" else "val"
        data = []

        with open(os.path.join(filepath, f"{filename}_lm.json")) as f:
            j = json.load(f)

        for i, (table_id, examples) in enumerate(j.items()):
            table = []
            with open(os.path.join(filepath, "all_csv", table_id)) as f:
                for line in f.readlines():
                    table.append(line.rstrip("\n").split("#"))

            for example in examples:
                data.append({
                    "table": table,
                    "ref": example[0],
                    "linked_columns": example[1],
                    "title": example[2],
                    "template": example[3],
                    "table_id": table_id,
                })
        for example_idx, entry in enumerate(data):
            yield example_idx, {key: str(value) for key, value in entry.items()}


if __name__ == '__main__':
    dataset = datasets.load_dataset(__file__)
    dataset.push_to_hub("kasnerz/logicnlg")
```
